saleor/graphql/product/mutations/products.py

```diff
@@ -439,6 +461,8 @@ def perform_mutation(cls, _root, info, collection_id, products):
         )
         products = cls.get_nodes_or_error(products, "products", only_type=Product)
         collection.products.remove(*products)
+        for product in products:
```

Comment:
Should we make queryset with prefetched needed data from products list?

Reply:
I added prefetch_related.